examples/platform/silabs/BaseApplication.cpp

```diff
@@ -458,11 +458,7 @@ void BaseApplication::ButtonHandler(AppEvent * aEvent)
             // - Cycle LCD screen
             CancelFunctionTimer();
 
-            OutputQrCode(false);
-#ifdef DISPLAY_ENABLED
-            UpdateLCDStatusScreen();
-            slLCD.CycleScreens();
-#endif
+            AppTask::GetAppTask().UpdateDisplay();
```

Comment:
When the child class creates a static method with the same signatures as the parent, it does no "overrides" the parent method, it actually overloads it.
If you call the static method from the object instance, as in this case, the code reads like inheritance, but it is not. This might cause unexpected behaviors.For instance:
```
#include <stdio.h>

struct Red
{
  Red(int y): x(y) {}
  static void print()
  {
    printf("Red!");
  }
  int x = 0;
};

struct Blue: public Red
{
  Blue(): Red(1) {}
  static void print()
  {
    printf("Blue");
  }
};

struct Green: public Red
{
  Green(): Red(2) {}
  static void print()
  {
    app_log_info("Green");
  }
};

void test(Red &r)
{
  printf("X: %d\n", r.x);
  r.print();
}

int main(void)
{
    Blue b;
    Green g;
    test(b);
    test(g);
    return 0;
}
```
You might expect this code to print:
```
X:1 Blue
X:2 Green
```
Instead, it prints:
```
X:1 Red!
X:2 Red!
```
The consequences might range to none to catastrophic, but you should avoid confusions either way.

Reply:
Changed the parent class to have a virtual method